File: frontend/src/types/dataset-version.ts
export type DatasetExportStatus = 'success' | 'exporting' | 'failed' | 'pending'

export interface DatasetVersionSummary {
  id: string
  datasetId: string
  datasetName: string
  versionName: string
  source: 'annotation-export' | 'imported' | 'manual-freeze'
  imageCount: number
  annotationCount: number
  classCount: number
  splitSummary: string
  exportFormat: string
  status: 'ready' | 'draft' | 'archived'
  createdBy: string
  createdAt: string
  notes: string
  splitDetail?: Array<{ split: string; images: number; annotations: number }>
  classDistribution?: Array<{ className: string; count: number; ratio: string }>
  validationHints?: string[]
}

export interface DatasetExportRecord {
  id: string
  datasetId: string
  datasetName: string
  versionName: string
  exportName: string
  exportFormat: 'yolo' | 'coco' | 'voc'
  exportScope: string
  status: 'completed' | 'processing' | 'failed'
  outputPath: string
  createdBy: string
  createdAt: string
  notes?: string
  includedAssets?: string[]
  validationSummary?: string[]
}

export interface DatasetVersionCreatePayload {
  dataset_id: string
  version_name: string
  description?: string
  source_type: 'annotation-export' | 'imported' | 'manual-freeze'
  export_format: 'yolo'
  include_splits: string[]
  split_strategy: string
  split_config?: Record<string, unknown>
}

export interface DatasetVersionValidationIssue {
  code: string
  message: string
  level: string
}

export interface DatasetExportRecordDetail {
  id: string
  dataset_id: string
  dataset_version_id: string
  export_name: string
  export_format: string
  status: DatasetExportStatus
  split_config: Record<string, unknown> | null
  output_path: string | null
  data_yaml_path: string | null
  manifest_path: string | null
  validation_summary: {
    passed: boolean
    errors: DatasetVersionValidationIssue[]
    warnings: DatasetVersionValidationIssue[]
    summary: Record<string, unknown>
  } | null
  error_message: string | null
  finished_at: string | null
  created_at: string
  updated_at: string
}

export interface DatasetVersionValidationResult {
  passed: boolean
  errors: DatasetVersionValidationIssue[]
  warnings: DatasetVersionValidationIssue[]
  summary: Record<string, unknown>
}

export interface DatasetVersionApiRecord {
  id: string
  dataset_id: string
  version_name: string
  version_code: string
  description: string | null
  status: string
  source_type: string
  export_format: string
  include_splits: string[] | null
  split_strategy: string | null
  split_config: Record<string, unknown> | null
  label_schema_snapshot: Array<Record<string, unknown>> | null
  stats_snapshot: Record<string, unknown> | null
  validation_summary: {
    passed: boolean
    errors: DatasetVersionValidationIssue[]
    warnings: DatasetVersionValidationIssue[]
    summary: Record<string, unknown>
  } | null
  frozen_at: string | null
  created_at: string
  updated_at: string
}

export interface DatasetVersionPlanItem {
  title: string
  description: string
}

export interface DatasetVersionOverviewCard {
  label: string
  value: string | number
  help: string
}

export interface ExportDraft {
  exportName: string
  exportFormat: 'yolo' | 'coco' | 'voc'
  splits: string[]
  extras: string[]
  outputPath: string
  notes: string
}

File: backend/app/schemas/dataset_version.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class DatasetVersionCreate(BaseModel):
    dataset_id: uuid.UUID
    version_name: str
    description: str | None = None
    source_type: str = "manual-freeze"
    export_format: str = "yolo"
    include_splits: list[str] = Field(default_factory=lambda: ["train", "val", "test"])
    split_strategy: str = "reuse-existing"
    split_config: dict[str, Any] | None = None


class DatasetVersionValidationIssue(BaseModel):
    code: str
    message: str
    level: str


class DatasetVersionValidationResult(BaseModel):
    passed: bool
    errors: list[DatasetVersionValidationIssue]
    warnings: list[DatasetVersionValidationIssue]
    summary: dict[str, Any]


class DatasetVersionResponse(BaseModel):
    id: uuid.UUID
    dataset_id: uuid.UUID
    version_name: str
    version_code: str
    description: str | None
    status: str
    source_type: str
    export_format: str
    include_splits: list[str] | None
    split_strategy: str | None
    split_config: dict[str, Any] | None
    label_schema_snapshot: list[dict[str, Any]] | None
    stats_snapshot: dict[str, Any] | None
    validation_summary: dict[str, Any] | None
    frozen_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DatasetExportCreate(BaseModel):
    dataset_version_id: uuid.UUID
    export_name: str
    export_format: str = "yolo"
    splits: list[str] = Field(default_factory=lambda: ["train", "val", "test"])
    extras: list[str] = Field(
        default_factory=lambda: ["include_images", "include_labels", "include_manifest"]
    )
    notes: str | None = None


class DatasetExportResponse(BaseModel):
    id: uuid.UUID
    dataset_id: uuid.UUID
    dataset_version_id: uuid.UUID
    export_name: str
    export_format: str
    status: str
    split_config: dict[str, Any] | None
    output_path: str | None
    data_yaml_path: str | None
    manifest_path: str | None
    validation_summary: dict[str, Any] | None
    error_message: str | None
    finished_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
